[2024-06-01T16:13:11.573+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-01T16:13:11.637+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_dag.transform scheduled__2023-03-22T00:00:00+00:00 [queued]>
[2024-06-01T16:13:11.648+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_dag.transform scheduled__2023-03-22T00:00:00+00:00 [queued]>
[2024-06-01T16:13:11.648+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-06-01T16:13:11.668+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): transform> on 2023-03-22 00:00:00+00:00
[2024-06-01T16:13:11.712+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=332) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-06-01T16:13:11.713+0000] {standard_task_runner.py:63} INFO - Started process 338 to run task
[2024-06-01T16:13:11.706+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'etl_dag', 'transform', 'scheduled__2023-03-22T00:00:00+00:00', '--job-id', '171', '--raw', '--subdir', 'DAGS_FOLDER/etl_dag.py', '--cfg-path', '/tmp/tmp8e9bjtg6']
[2024-06-01T16:13:11.720+0000] {standard_task_runner.py:91} INFO - Job 171: Subtask transform
[2024-06-01T16:13:11.762+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2024-06-01T16:13:11.817+0000] {task_command.py:426} INFO - Running <TaskInstance: etl_dag.transform scheduled__2023-03-22T00:00:00+00:00 [running]> on host c5ef054da447
[2024-06-01T16:13:11.943+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='etl_dag' AIRFLOW_CTX_TASK_ID='transform' AIRFLOW_CTX_EXECUTION_DATE='2023-03-22T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2023-03-22T00:00:00+00:00'
[2024-06-01T16:13:11.947+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-01T16:13:12.324+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-01T16:13:12.326+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/etl_dag.py", line 42, in transform
    pdf_link = soup.find("a", title="Sheet nhạc bài Chờ ngày lời hứa nở hoa")['href']  # Adjust selector
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
[2024-06-01T16:13:12.415+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=etl_dag, task_id=transform, run_id=scheduled__2023-03-22T00:00:00+00:00, execution_date=20230322T000000, start_date=20240601T161311, end_date=20240601T161312
[2024-06-01T16:13:12.452+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 171 for task transform ('NoneType' object is not subscriptable; 338)
[2024-06-01T16:13:12.508+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-01T16:13:12.523+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
